### PR TITLE
Refine "REDACTED" replacement of API key value in web fetcher search URL

### LINKED ISSUE #13796: Refine "REDACTED" replacement
In case of an error, they key is removed from the output.

Example:

1. Enable Web Fetchers
2. Select "Springer"
3. Search for "anything"

You will get

Image

The output looks a bit strange as one would expect that only the key is not shown - and thus, the URL makes the impression of being invalid

## Task

1. Modify the constant to have a regex group for the API key: https://github.com/JabRef/jabref/blob/1aae72d2c61c702038c5a67b6ed8d4805df4a3ba/jablib/src/main/java/org/jabref/logic/importer/FetcherException.java#L17. Use a "named group" for readability.
2. Modify `getRedactedUrl` (https://github.com/JabRef/jabref/blob/1aae72d2c61c702038c5a67b6ed8d4805df4a3ba/jablib/src/main/java/org/jabref/logic/importer/FetcherException.java#L89) to replace only the group of the key - and not all.
3. Make the method returning `String` instead of Object.
4. Make the method package private and add `@VisibleForTesting`
5. Add test cases - using `@ParameterizedTest` and `@CsvSource`. Hint: Use <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>T</kbd> to create a test class.

To increase debugging - the first letter of the API key could still be output - please modify the regex accordingly.

## Benefits

Learn how regular expressions work in Java

### PR DESCRIPTION
**Pull request by josephyim**
Closes #13796

Improve URL redaction: change the regex to redact only the API key's value (not the whole name=value). Add tests to cover common key names and edge cases.

### Steps to test
In Web search, choose "Springer", search anything, then you will see the error message below:
image

### Mandatory checks
- [x] I own the copyright of the code submitted and I license it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] I manually tested my changes in running JabRef (always required)
- [x] I added JUnit tests for changes (if applicable)
- [x] I added screenshots in the PR description (if change is visible to the user)
- [x] I described the change in `CHANGELOG.md` in a way that is understandable for the average user (if change is visible to the user)
- [x] I checked the [user documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request updating file(s) in <https://github.com/JabRef/user-documentation/tree/main/en>.

### CODE REVIEW

File: jablib/src/main/java/org/jabref/logic/net/URLDownload.java [hunk 1]
```diff
@@ -382,7 +382,7 @@ public URLConnection openConnection() throws FetcherException {
             } else if (status >= 400) {
                 // in case of an error, propagate the error message
                 SimpleHttpResponse httpResponse = new SimpleHttpResponse(httpURLConnection);
-                LOGGER.info("{}: {}", FetcherException.getRedactedUrl(this.source), httpResponse);
+                LOGGER.info("{}: {}", FetcherException.getRedactedUrl(this.source.toString()), httpResponse);
```
Comment (koppor) [inline jablib/src/main/java/org/jabref/logic/net/URLDownload.java:385]: Isn't `toString()` automatically called when logging?

Comment (Siedlerchr) [reply jablib/src/main/java/org/jabref/logic/net/URLDownload.java:385]: Yes

**Review by koppor — APPROVED** [review]
